R/util_assert.R

```diff
@@ -80,6 +80,37 @@ assert_file_exists <- function(files, name = "File", call = NULL, arg = NULL) {
 }
 
 
+find_entrypoint_filename <- function(src, reportname,
+                                       suppress_errors = FALSE) {
+  names <- c(sprintf("%s.R", reportname), "orderly.R")
+  files_exist <- file.exists(file.path(src, names))
+  if (sum(files_exist) > 1 && !suppress_errors) {
+    cli::cli_abort(
+      paste("Please only create {names[[1]]} file, orderly.R",
+            "has been deprecated")
+    )
+  } else if (sum(files_exist) == 0 && !suppress_errors) {
+    cli::cli_abort(
+      "Please create {names[[1]]} file"
+    )
+  } else if (files_exist[[2]]) {
+    rlang::inform(
+      paste("Naming convention orderly.R will be deprecated",
+            "soon. Please change orderly file name to",
+            "<reportname>.R"),
+      .frequency = "regularly",
+      .frequency_id = "deprecate_orderly_file_name"
+    )
+  }
+
+  if (suppress_errors && sum(files_exist) != 1) {
+    NULL
+  } else {
+    names[files_exist]
+  }
```

Comment:
```suggestion
  if (sum(files_exist) == 1) names[files_exist] else NA_character_
```

* no need to test `suppress_errors`
* see above for suggestion to replace return type with missing string rather than `NULL`
* can do it on one line, which I find sometimes clearer

Also you sum files_exist several times, here could pull that into `n_found <- sum(files_exist)` if you want

Reply:
yh looks much cleaner, done